examples/stm32/xively_demo_stm32f4xx_nucleo_eth/SW4STM32/syscalls.c

```diff
@@ -133,14 +133,9 @@ int _lseek(int file, int ptr, int dir)
 
 int _read(int file, char *ptr, int len)
 {
-	int DataIdx;
-
-	for (DataIdx = 0; DataIdx < len; DataIdx++)
-	{
-	  *ptr++ = __io_getchar();
-	}
-
-   return len;
+    /* scanf calls _read() with len=1024, so eat one character at time */
+    *ptr = __io_getchar();

```

Comment:
I copied that comment and implementation from the wifi's `syscalls.c`. I'd also like to know why this implementation can ignore the `len` param, but it fixed all my issues, so I didn't investigate further.